soroban-sdk/src/crypto/poseidon2_params.rs

```diff
@@ -0,0 +1,346 @@
+use crate::{bytesn, vec, Env, Vec, U256};
+
+pub const T: usize = 4;
+pub const SBOX_D: u32 = 5;
+pub const ROUNDS_F: u32 = 8;
+pub const ROUNDS_P: u32 = 56;
+
+// These parameters matches reference parameters generated with the reference sage script (t = 4, field = BN254):
+// https://github.com/HorizenLabs/poseidon2/blob/055bde3f4782731ba5f5ce5888a440a94327eaf3/poseidon2_rust_params.sage
+#[rustfmt::skip]
+pub fn get_mat_diag4_m_1(e: &Env) -> Vec<U256> {
+    vec![
+        e,
+        U256::from_be_bytes(e, &bytesn!(e, 0x10dc6e9c006ea38b04b1e03b4bd9490c0d03f98929ca1d7fb56821fd19d3b6e7).into()),
+        U256::from_be_bytes(e, &bytesn!(e, 0x0c28145b6a44df3e0149b3d0a30b3bb599df9756d4dd9b84a86b38cfb45a740b).into()),
+        U256::from_be_bytes(e, &bytesn!(e, 0x00544b8338791518b2c7645a50392798b21f75bb60e3596170067d00141cac15).into()),
+        U256::from_be_bytes(e, &bytesn!(e, 0x222c01175718386f2e2e82eb122789e352e105a3b8fa852613bc534433ee428b).into()),
+    ]
+}
+
+#[rustfmt::skip]
+pub fn get_rc4(e: &Env) -> Vec<Vec<U256>>
```

Comment:
My questions about this are likely to be naive. But wow there will be a lot of host function calls for this fn, that gets called once for each new `Poseidon2Sponge`. Is it likely or possible someone will need to create multiple `Poseidon2Sponge`? Maybe we should find a way to do `get_rc4` once only.